cpp/Makefile

```diff
@@ -1,7 +1,8 @@
 # Build a shared library of negentropy
 
-#TODO: Need to add compilation flags based on OS
-INCS	= -I./ -I/opt/homebrew/include/ -I./vendor/lmdbxx/include/
+# Define the root directory of the negentropy project; this absolute path mechanism works across all major os
+NEGENTROPY_ROOT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+INCS	= -I$(NEGENTROPY_ROOT) -I/opt/homebrew/include/ -I$(NEGENTROPY_ROOT)/vendor/lmdbxx/include/
 TARGET  = libnegentropy.so
```

Comment:
shouldn't this be .dll for it to work in windows?
not sure if .so format is recognized by windows.